src/library/scala/util/control/Exception.scala

```diff
@@ -13,24 +13,125 @@ package control
 import scala.reflect.{ ClassTag, classTag }
 import scala.language.implicitConversions
 
-
 /** Classes representing the components of exception handling.
- *  Each class is independently composable.  Some example usages:
+ *
+ *  Each class is independently composable.
+ *
+ *  This class differs from [[scala.util.Try]] in that it focuses on composing exception handlers rather than
+ *  composing behavior.  All behavior should be composed first and fed to a [[Catch]] object using one of the
+ *  `opt`, `either` or `withTry` methods. Taken together the classes provide a DSL for composing catch and finally
+ *  behaviours.
+ *
+ *  === Examples ===
+ *
+ *  Create a `Catch` which handles specified exceptions.
  *  {{{
  *  import scala.util.control.Exception._
  *  import java.net._
  *
  *  val s = "http://www.scala-lang.org/"
+ *
+ *  // Option[URL]
```

Comment:
I too have read a suggestion that examples should be copy/paste/runnable in the REPL. I started with this approach and I liked it but using an IDEA worksheet was easier in the end. I'll try it again now the examples are stable and see what the output looks like. Failing that moving the types inline is a good idea.